### ISSUE TITLE
Allow different marker symbols for all points in MarkerVisual

### ISSUE THREAD

**Issue by rossant** [vispy/vispy#753]
Instead of just passing a string like `^` or `o` as `symbol`, pass a list or an array of such symbols, with one value per point.

**Comment by rossant** [comment]
More generally I'm wondering if the collections system can buy us an automatic system that allows for heterogeneous values for all visual properties.

**Comment by rougier** [comment]
Yes and no. Yes for most properties but marker type requires some efforts because a type=dedicated code. And since we do not have function pointers, we have to generate a piece of code that minimise the number of test s(no need to test sequentially against all marker types if we know before how many different marker type we have in a collection).

**Comment by rossant** [comment]
OK so we'll have to write custom code for markers, but for the other properties it would be nice to have an automatic system eventually. We can discuss that during the sprint.

**Comment by queenish001** [comment]
@rossant Is this issue resolved?

**Comment by djhoese** [comment]
This is still a valid request and can stay open. I think it could be possible to build a proxy function that maps a "marker_type" integer to an integer, that integer goes into a buffer, and then there's a nice `if` statement to choose what marker is used for each vertex. It shouldn't perform too bad until you try to use a ton of different types of markers and since the Function object is created when the symbol(s) are set then any performance issues can be avoided (no ever-growing `if` statement in the shader that effects all cases).

At least this should work off the top of my head.

**Comment by rougier** [comment]
If we use SDF markers (signed distance fields), then we have a texture atlas gathering all the different markers at once and we can have different marker for each point Since they are markers, the resolution for each marker do not need to be too big (eg 64x64). For a given collection, it would only a matter of giving the proper marker index. Only drawback is that for very big marker, quality would suffer a bit compared to the pure mathematical marker (i.e. code snippet).
